crates/block-producer/src/test_utils/block.rs

```diff
@@ -22,7 +22,14 @@ pub async fn build_expected_block_header(
     store: &MockStoreSuccess,
     batches: &[TransactionBatch],
 ) -> BlockHeader {
-    let last_block_header = *store.last_block_header.read().await;
+    let last_block_header = *store
+        .block_headers
+        .read()
+        .await
+        .iter()
+        .max_by_key(|(block_num, _)| *block_num)
+        .unwrap()
+        .1;
```

Comment:
This is really awkward to do everywhere, but I didn't think its safe to extract this as a method since it holds a lock. So calling it elsewhere could easily result in an accidental dead-lock.